casanovo/data/db_utils.py

```diff
@@ -0,0 +1,389 @@
+        precursor_mz : float
+            The precursor mass-to-charge ratio.
+        charge : int
+            The precursor charge.
+
+        Returns
+        -------
+        candidates : pd.Series
+            A series of candidate peptides.
+        """
+        candidates = []
+
+        for e in range(self.isotope_error[0], self.isotope_error[1] + 1):
+            iso_shift = ISOTOPE_SPACING * e
+            shift_raw_mass = float(
+                _to_raw_mass(precursor_mz, charge) - iso_shift
+            )
+            upper_bound = shift_raw_mass * (
+                1 + (self.precursor_tolerance / 1e6)
+            )
+            lower_bound = shift_raw_mass * (
+                1 - (self.precursor_tolerance / 1e6)
+            )
+
+            window = self.db_peptides[
+                (self.db_peptides["calc_mass"] >= lower_bound)
+                & (self.db_peptides["calc_mass"] <= upper_bound)
+            ]
+            candidates.append(window[["peptide", "calc_mass", "protein"]])
+
+        candidates = pd.concat(candidates)
+        candidates.drop_duplicates(inplace=True)
+        candidates.sort_values(by=["calc_mass", "peptide"], inplace=True)
+        return candidates["peptide"], candidates["protein"]
+
+    def get_associated_protein(self, peptide: str) -> str:
+        """
+        Returns the associated protein for a given peptide.
+
+        Parameters
+        ----------
+        peptide : str
+            The peptide sequence.
+
+        Returns
+        -------
+        protein : str
+            The associated protein(s).
+        """
+        return ",".join(self.prot_map[peptide])
+
+    def _digest_fasta(
+        self,
+        fasta_filename: str,
+        enzyme: str,
+        digestion: str,
+        missed_cleavages: int,
+        max_mods: int,
+        min_peptide_length: int,
+        max_peptide_length: int,
+    ) -> pd.DataFrame:
+        """
+        Digests a FASTA file and returns the peptides, their masses, and associated protein.
+
+        Parameters
+        ----------
+        fasta_filename : str
+            Path to the FASTA file.
+        enzyme : str
+            The enzyme to use for digestion.
+            See pyteomics.parser.expasy_rules for valid enzymes.
+            Can also be a regex pattern.
+        digestion : str
+            The type of digestion to perform. Either 'full', 'partial' or 'non-specific'.
+        missed_cleavages : int
+            The number of missed cleavages to allow.
+        max_mods : int
+            The maximum number of modifications to allow per peptide.
+        min_peptide_length : int
+            The minimum length of peptides to consider.
+        max_peptide_length : int
+            The maximum length of peptides to consider.
+
+        Returns
+        -------
+        pep_table : pd.DataFrame
+            A Pandas DataFrame with peptide, mass,
+            and protein columns. Sorted by neutral mass in ascending order.
```

Comment:
**question:** Why does the protein column still need to be in here?

Reply:
Removed